src/mongocxx/validation_criteria.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mongocxx/validation_criteria.hpp>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+
+#include <mongocxx/config/prelude.hpp>
+
+namespace {
+
+using namespace mongocxx;
+// using mongocxx::validation_criteria::validation_action;
+// using mongocxx::validation_criteria::validation_level;
+
+///
+/// Convert validation levels to strings.
+///
+std::string validation_level_to_string(validation_criteria::validation_level level) {
+    switch (level) {
+        case validation_criteria::validation_level::k_off:
+            return "off";
+        case validation_criteria::validation_level::k_moderate:
+            return "moderate";
+        case validation_criteria::validation_level::k_strict:
+            return "strict";
+    }
+}
+
+///
+/// Convert validation actions to strings.
+///
+std::string validation_action_to_string(validation_criteria::validation_action action) {
+    switch (action) {
+        case validation_criteria::validation_action::k_warn:
+            return "warn";
+        case validation_criteria::validation_action::k_error:
+            return "error";
+    }
+}
+
+}  // namespace
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+void validation_criteria::rule(bsoncxx::document::view_or_value rule) {
+    _rule = std::move(rule);
+}
+
+void validation_criteria::level(validation_criteria::validation_level level) {
+    _level = std::move(level);
+}
+
+void validation_criteria::action(validation_criteria::validation_action action) {
+    _action = std::move(action);
+}
+
+bsoncxx::document::value validation_criteria::to_document() const {
+    auto doc = bsoncxx::builder::stream::document{};
+
+    if (_rule) {
+        doc << "validator" << bsoncxx::types::b_document{*_rule};
```

Comment:
The stream builder doesn't currently have an overload to take document::views as value types.  It would be nice if it did, I've opened [CXX-784](https://jira.mongodb.org/browse/CXX-784) so we can discuss it there.